tests/src/JIT/jit64/mcc/interop/native_i1c.cpp

```diff
@@ -4,25 +4,20 @@
 #include <stdarg.h>
 #include "native.h"
 
-#ifdef __clang__
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wvarargs"
-#endif
-
-MCC_API VType1 sum(float first, ...) {
+MCC_API VType1 sum(double first, ...) {
     VType1 result;
 
     int count = 0;
     float sum = 0.0;
-    float val = first;
+    float val = (float)first;
```

Comment:
Using float in this case should be undefined behavior. Can you remove the casts and just use doubles. We should use the promoted type. On the caller side we should also make sure to pass double. I believe this test is still valid for full framework. I believe we need to support passing unwidened types for backcompat.

Reply:
Also change `float` to `double` in [`struct VType1`](https://github.com/dotnet/coreclr/blob/0495f7377fca88999d335757e9680052cfedc51d/tests/src/JIT/jit64/mcc/interop/native.h#L39)?

Reply:
The return type seems fine to me

Reply:
FYI, This ended up breaking the tests (see #22974) as we don't do the expected promotion of float to double for floats passed via native varargs on the managed side.

@jashook not sure what you meant by

> On the caller side we should also make sure to pass double.